Notification Settings: Add track events to Notification > Extras  page

gabrielcaires opened the pull request:
Closes CML-942

## Proposed Changes
* Identify if the settings changes are coming from the "Subscribe/Unsubscribe all" 
* Identify the group the settings were enabled/disabled (wpcom or jetpack)

* New events
* `calypso_dashboard_notifications_settings_subscribe_all`
* `calypso_dashboard_notifications_settings_unsubscribe_all`  
* `calypso_dashboard_notifications_settings_subscribe_update`  --> It is always triggered with the settings name and settings value.

The legacy version was not tracking it using track event, only Google Analytics.


## Testing Instructions
* Go to `v2/me/notifications/extras`
* Click on the Subscribe/Unsubscribe toggle 
* Use Vigilant tracks to check if the events were triggered properly


<img width="1124" height="334" alt="CleanShot 2025-10-03 at 15 49 56@2x" src="https://github.com/user-attachments/assets/d4b56c8f-9850-4405-aa4c-d15455e19c3d" />
<img width="1276" height="334" alt="CleanShot 2025-10-03 at 15 50 48@2x" src="https://github.com/user-attachments/assets/e60c36c9-af36-4bd3-b8dc-79f33889e0c5" />
<img width="1270" height="284" alt="CleanShot 2025-10-03 at 15 51 37@2x" src="https://github.com/user-attachments/assets/f53ed8c6-8261-4d66-b538-eaf23d14fd8d" />



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gabrielcaires committed:
Add tracking

gabrielcaires committed:
Always trigger the single event

gabrielcaires committed:
Improve event name

gabrielcaires commented:
I updated the event name, so the screenshots are outdated

jeherve reviewed (commented):
When I click to unsubscribe from all (or subscribe to all), the `_all` events are properly triggered but a `single` event is also triggered:

<img width="1290" height="99" alt="Image" src="https://github.com/user-attachments/assets/41436b4f-fb6d-4a59-a306-c3acc790f478" />

That seems like something we should avoid.

gabrielcaires commented:
@jeherve In this case, I think we should track both cases, because if we build a report with the number of times where setting X was disabled, it, in theory, should not matter if it was triggered by the click or the unsubscribe/subscribe all. 

What do you think about it?

gabrielcaires committed:
Identify when a proper was changes via subscribe all on via single settings

gabrielcaires commented:
> Ideally I think we should track the event, but maybe we should be able to say if it was a direct click or an indirect event (triggered via the "unsubscribe all").

@jeherve  I renamed the single to be just `calypso_dashboard_notifications_settings_subscribe_update`  to follow the same idea of other PRs, and I added a prop to represent if it was updated by subscribe/unsubscribe all or individually.

gabrielcaires committed:
Merge branch 'trunk' into update/cml-942-add-tracking-to-notification-settings-page

jeherve reviewed (approved):
Works well for me, I like that approach.
